src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -192,44 +197,55 @@ internal async Task<bool> UpdateApplication(TokenCredential tokenCredential, App
                 //update redirect uris
                 List<string> existingRedirectUris = updatedApp.Web.RedirectUris.ToList();
                 List<string> urisToEnsure = ValidateUris(toolOptions.RedirectUris).ToList();
-                existingRedirectUris.AddRange(urisToEnsure);
-                updatedApp.Web.RedirectUris = existingRedirectUris.Distinct();
-
-                //update implicit grant settings
+                List<string> urisToAdd = existingRedirectUris.Except(urisToEnsure).ToList();
+                if (urisToAdd.Any())
+                {
+                    needsUpdate = true;
+                    existingRedirectUris.AddRange(urisToAdd);
+                    updatedApp.Web.RedirectUris = existingRedirectUris.Distinct();
+                }
+                
                 if (updatedApp.Web.ImplicitGrantSettings == null)
                 {
                     updatedApp.Web.ImplicitGrantSettings = new ImplicitGrantSettings();
                 }
 
-                if (toolOptions.EnableAccessToken.HasValue)
+                //update implicit grant settings if need be.
+                if (toolOptions.EnableAccessToken.HasValue && (toolOptions.EnableAccessToken.Value != updatedApp.Web.ImplicitGrantSettings.EnableAccessTokenIssuance))
                 {
+                    needsUpdate = true;
                     updatedApp.Web.ImplicitGrantSettings.EnableAccessTokenIssuance = toolOptions.EnableAccessToken.Value;
                 }
 
-                if (toolOptions.EnableIdToken.HasValue)
+                if (toolOptions.EnableIdToken.HasValue && (toolOptions.EnableIdToken.Value != updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance))
```

Comment:
Here, HasValue rather than GetValueOrDefault is imporant :-)

Reply:
yup makes sense.